src/artifacts.js

```diff
@@ -4,8 +4,41 @@ let assert  = require('assert');
 let Promise = require('promise');
 let API     = require('taskcluster-lib-api');
 let urllib  = require('url');
+let crypto  = require('crypto');
 let api     = require('./api');
 
+/**
+ * Encode a storage key for storing in the blob storage type backends We do
+ * this to get better partitioning in S3 which gives us better performance.
+ *
+ * Basically `${md5hash(taskId + runId + name).slice(0,7)}/${taskId}/${runId}/${name}`
```

Comment:
We could always reconstruct it by listing the objects in the buckets and then trimming off the first path, but I think you're right that random taskIds should be enough randomness for the partitioning impact.  I'd just forgotten about that detail :)